### PR TITLE
Bug 1881208: add alert for system.slice rss going over system-reserved

### PR DESCRIPTION
**Pull request by sjenning**
@smarterclayton @mrunalp @nee1esh @derekwaynecarr 

xref https://issues.redhat.com/browse/OCPNODE-402

### CODE REVIEW

**Comment by mrunalp** [comment]
I think we want to get a number that accounts for pages that we aren't able to evict from cri-o on drop caches. Also, the alert could be at 90 -95% of the reservation.

File: install/0000_90_machine-config-operator_01_prometheus-rules.yaml [hunk 1]
```diff
@@ -43,3 +43,12 @@ spec:
             severity: warning
           annotations:
             message: "Kubelet health failure threshold reached"
+    - name: system-memory-exceeds-reservation
+      rules:
+        - alert: SystemMemoryExceedsReservation
+          expr: |
+            container_memory_rss{id="/system.slice"} > 1073741824
```
Comment (smarterclayton) [inline install/0000_90_machine-config-operator_01_prometheus-rules.yaml:50]: Hrm, this needs to come from the reservation the system makes, which is exposed as a metric

Comment (smarterclayton) [reply install/0000_90_machine-config-operator_01_prometheus-rules.yaml:50]: constant should be kube_node_status_capacity{... mem} - kube_node_status_allocatable{... mem}, and you need to figure out how to make that number show up in the annotation somehow

Comment (smarterclayton) [reply install/0000_90_machine-config-operator_01_prometheus-rules.yaml:50]: is there anything on a node outside of system and kubepods?  should the first value be `container_memory_rss{id="/"} - container_memory_rss{id="/kubepods.slice"} > (capacity - allocatable)`

Comment (sjenning) [reply install/0000_90_machine-config-operator_01_prometheus-rules.yaml]: all the kernel threads are in `/`. no userspace process though.

File: install/0000_90_machine-config-operator_01_prometheus-rules.yaml [hunk 2]
```diff
@@ -43,3 +43,12 @@ spec:
...
             message: "Kubelet health failure threshold reached"
+    - name: system-memory-exceeds-reservation
+      rules:
+        - alert: SystemMemoryExceedsReservation
+          expr: |
+            sum by (node) (container_memory_rss{id="/system.slice"}) > ((sum by (node) (kube_node_status_capacity{resource="memory"} - kube_node_status_allocatable{resource="memory"})) * 0.9)
+          labels:
+            severity: warning
+          annotations:
+            message: "System memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 90% of the reservation"
```
Comment (smarterclayton) [inline install/0000_90_machine-config-operator_01_prometheus-rules.yaml:54]: Messages should be acitonable.  So you need to recommend next steps.  I would suggest playing around with something like the following additions:

> Reserved memory ensures system processes can function even when the node is fully allocated and protects against workload out of memory events impacting the proper functioning of the node. The reservation may be increased (link to docs) when running nodes with high numbers of pods.

Or similar

Comment (sjenning) [reply install/0000_90_machine-config-operator_01_prometheus-rules.yaml]: @smarterclayton better now?

**Comment by smarterclayton** [comment]
/lgtm

Attach the bug and this is gtg for 4.6

**Review by kikisdeliveryservice — APPROVED** [review]

**Pull request merged**